internal/uvm/hvsocket.go

```diff
@@ -0,0 +1,45 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// UpdateHvSocketService calls HCS to update/create the hvsocket service for
+// the UVM. Takes in a service ID and the hvsocket service configuration. If there is no
+// entry for the service ID already it will be created. The same call on HvSockets side
+// handles the Create/Update/Delete cases based on what is passed in. Here is the logic
+// for the call.
+//
+// 1. If the service ID does not currently exist in the service table, it will be created
+// with whatever descriptors and state was specified (disabled or not).
+// 2. If the service already exists and empty descriptors and Disabled is passed in for the
+// service config, the service will be removed.
+// 3. Otherwise any combination that is not Disabled && Empty descriptors will just update the
+// service.
+//
+// This function should preferably be called for the Update/Create flow but there isn't anything
+// stopping this being used for the delete case. Prefer RemoveHvSocketService as this sets the
+// relevant fields for the Delete case on HCS' side.
```

Comment:
Why would we potentially want to use this in the delete case? Why not just use the preferred "RemoveHvSocketService"?

Reply:
There isn't really a reason that I can think of. I'm just simply stating that nothing is stopping the caller from crafting a request that ends up disabling the service as they all get routed to the same hvsocket call at the end. Is there a way you want this worded differently maybe?

Reply:
I think just removing that part would be better so as not to confuse the reader. It should be clear that we want to use the remove func for removal.